libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/js/npm/tasks/KotlinNpmInstallTask.kt

```diff
@@ -41,8 +41,15 @@ abstract class KotlinNpmInstallTask :
     @get:Internal
     abstract val nodeModules: DirectoryProperty
 
+    private val isOffline = project.gradle.startParameter.isOffline()
+
     @TaskAction
     fun resolve() {
+        val args = buildList {
+            addAll(args)
+            if (isOffline) add("--offline")
```

Comment:
May I also ask you to introduce a test for this logic? 
@ilgonmic, is it possible to somehow emulate the "offline" mode in our Gradle integration test infrastructure?

Reply:
Yep, I would love to cover this with tests, but where can I see some examples?

Reply:
We have integration tests for the Gradle plugin. You can explore the JS-specific here: https://github.com/JetBrains/kotlin/blob/master/libraries%2Ftools%2Fkotlin-gradle-plugin-integration-tests%2Fsrc%2Ftest%2Fkotlin%2Forg%2Fjetbrains%2Fkotlin%2Fgradle%2FKotlin2JsGradlePluginIT.kt

The rest of the integration tests are located in the same Gradle sub-project